### PR TITLE
Add rest server docker image creation

### PR DESCRIPTION
**Pull request by mainnet-pat**
Check out the README.md on how to run 3 scenarios: http, https with own certificates and https with letsencrypt.

### CODE REVIEW

**Comment by 2qx** [comment]
Would it be possible to add a basic test in CI to assure this starts as expected?

**Comment by mainnet-pat** [comment]
Yes for the basic http and https with self-signed certificates is totally possible. Letsencrypt is a bit trickier, since it has rate limiter of 5 certs per week.

**Comment by mainnet-pat** [comment]
I'd have the CI in a separate changeset. As of this one, all checks are passing https://github.com/mainnet-pat/mainnet-js/actions/runs/360237099

File: generated/serve/docker/https.conf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+server {
+    listen 80;
+    return 301 https://$host$request_uri;
```
Comment (2qx) [inline generated/serve/docker/https.conf:3]: nice

**Comment by 2qx** [comment]
Okay, I'll go ahead and merge this.

I'm not familiar configuring letsencrypt for a single instance, but this looks good.

**Pull request merged**